packages/gatsby-theme-newrelic/src/components/GlobalHeader.js

```diff
@@ -27,7 +30,20 @@ const styles = {
 };
 
 const GlobalHeader = ({ editUrl, className, search }) => {
-  const [isOverlayOpen, setIsOverlayOpen] = useState(false);
+  const [isOverlayOpen, setIsOverlayOpen] = useState(null);
+
+  const [isQueryChanged, setIsQueryChanged] = useState(false);
+
+  useEffect(() => {
+    let searchQuery = qs.parse(window.location.search);
+    if (isOverlayOpen === null && searchQuery.overlay) setIsOverlayOpen(true);
+    if (isOverlayOpen === null && !searchQuery.overlay) setIsOverlayOpen(false);
+
+    if (isOverlayOpen && window.location.search.length && !searchQuery.overlay)
+      setQueryStringWithoutPageReload(window.location.search + '&overlay=true');
```

Comment:
I assume the `overlay=true` is meant to ensure the _search_ overlay is open when someone has searched correct? Since this is a pretty generic name, and isn't specific to search, I suggest 1 of 2 changes here. Pick your favorite.

Option 1: Rename the variable to something more specific to search (i.e. searchOverlay or something). 
Option 2: Open the overlay based on the existence of the search parameter. For example, if a search term adds a `q` query parameter to the URL, open the overlay if `q` is in the URL (even if the value is blank).

I like option 2 personally because it means you only need a single query string variable to determine whether to open the overlay instead of 2. This makes URL sharing a bit easier.

Reply:
The issue with this if we use the search component outside of the overlay, which what I was thinking about. If someone uses the global header and then decides to put search in the body, then they would be taken to the overlay when they don't necessarily want to.